packages/smart-contracts/packages/vesting/contracts/Vesting.sol

```diff
@@ -644,16 +638,11 @@ contract Vesting is IVesting, Initializable, AccessControlUpgradeable {
             Pool storage pool = s_vestingPools[i];
             Beneficiary storage user = pool.beneficiaries[msg.sender];
 
-            // Available tokens are the maximum amount that user should be able claim
-            // if all tokens are unlocked for the user,
-            uint256 availableTokens = user.totalTokenAmount -
-                user.claimedTokenAmount -
-                user.stakedTokenAmount;
-
             // Checks: Ensure the unlocked tokens do not exceed the available tokens
-            uint256 claimableTokens = unlockedTokens > availableTokens
-                ? availableTokens
-                : unlockedTokens;
+            uint256 claimableTokens = _calculateClaimableTokens(
+                user,
+                unlockedTokens
+            );
```

Comment:
If `claimableTokens` is 0, continue the loop with next item and skip the remaining lines for current item.